### PR TITLE
modify make targets

### PR DESCRIPTION
**Pull request by gyorb**
* package target creates a clean package (removes every dependecy and
  downloads them again)
* the package needs to be build before the tests explicitly, the
  test targets will not build the package

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -34,10 +34,8 @@ thrift: build_dir
 userguide: build_dir
 	$(MAKE) -C www/userguide
 
-package: build_dir gen-docs thrift userguide build_plist_to_html
-	if [ ! -d "$(BUILD_DIR)/CodeChecker" ]; then \
-		./scripts/build_package.py -r $(ROOT) -o $(BUILD_DIR) -b $(BUILD_DIR); \
-	fi
+package: clean_package build_dir gen-docs thrift userguide build_plist_to_html
+		./scripts/build_package.py -r $(ROOT) -o $(BUILD_DIR) -b $(BUILD_DIR)
```
Comment (csordasmarton) [inline Makefile]: Too much indentation.

File: tests/Makefile [hunk 2]
```diff
@@ -55,7 +55,7 @@ DROP_DB_CMD = bash -c 'psql -h localhost \
 		-p $${TEST_DBPORT} -U $${TEST_DBUSERNAME} postgres \
 		-c "DROP DATABASE codechecker_config_$${CODECHECKER_DB_DRIVER}"'
 
-run_test: package venv_dev
+run_test: venv_dev
```
Comment (csordasmarton) [inline tests/Makefile]: Why did you remove to build the package before tests?

Comment (gyorb) [reply tests/Makefile]: I removed them because we need to run the internal tests with a different CodeChecker configuration. If the test targets always rebuild the package before each test we can not run the tests with a different config.

File: Makefile (hunk 1 above)
Comment (gyorb) [reply Makefile]: fixed

File: tests/Makefile (hunk 2 above)
Comment (csordasmarton) [reply tests/Makefile]: In this case please update the documentation: https://github.com/Ericsson/codechecker/blob/master/tests/README.md

**Comment by whisperity** [comment]
@gyorb This will supersede #1109, right?

**Comment by gyorb** [comment]
@whisperity yes

File: tests/Makefile (hunk 2 above)
Comment (gyorb) [reply tests/Makefile]: done
